### ISSUE TITLE
Easier signup with third party

### ISSUE THREAD

**Issue by francois2metz** [openstreetmap/openstreetmap-website#2616]
Hi there,

I'm working on https://www.github.com/osmontrouge/caresteouvert and we would like to allows direct contributions to OpenStreetMap without using our proxy account to allow more specific edits (contacts, name, etc...).

For that, we would like a pretty straightforward way to create an account with third parties. Right now, once the authorization is done on the third party side, there is a page where we still have to fill our username and a password.

I'm thinking to remove this page so users can contribute right away:
- The email is automatically valided
- The username would be derived from the email or filled via the identity provider
- The password will be set to a random value (or empty) and if the user want to set it they use the forgot password workflow

I'm willing to implement theses changes if there is no objections on the idea.

**Comment by simonpoole** [comment]
@francois2metz this has been suggested before (specifically in the context of sign up from mobile devices).

The problems with providing an API for this (which technically is not difficult):

- messaging what OpenStreetMap is about (it would obviously be possible to provide the texts via an API),
- guaranteeing that the contributor terms and the terms of use have been presented to user and have actually been accepted in a valid fashion,
- e-mail address validation.

All the above is solvable, but would require putting a process in place to vet apps against a set of requirements that would guarantee the above, with regular checks that this is still done appropriately. I just don't see that happening without somebody specifically tasked with (and paid) for the work.

**Comment by francois2metz** [comment]
Thanks for the rapid answer that help me to be clearer about "third parties". I'm talking about the "Google/Facebook/Live/..." login and not about an API (which could be great, but a lot of works a you says).
Specificaly I'm talking about this page triggered via Log In -> third party login with Google:
![Screen Shot 2020-05-08 at 13 50 47](https://user-images.githubusercontent.com/86659/81403485-c6a0ad80-9133-11ea-8085-77d0edb7d7e6.png)


The email check it already done by Google/Facebook/Github/... so we may trust them about that.

**Comment by tomhughes** [comment]
Yes but after that page is another page where the user has to accept the terms and conditions.

We do already fill in the email address and username where they are available from the third party but we can't guarantee that the user won't need to change them - the username might already be in use for example.

So I think this is a non-starter. It is also a duplicate I think as something similar was asked before - not sure if it is still open or if it got closed.

**Comment by tomhughes** [comment]
Also while we can trust the email if it's a gmail address from authenticating via google that's not really true for the other providers as they don't control the email addresses so at best they know that the address was valid at some point in the past, so we still need to do an email confirmation.

**Comment by francois2metz** [comment]
> Yes but after that page is another page where the user has to accept the terms and conditions.

I think that we should keep it. The problem I want to solve is the need to enter information that we can derive from the third party authentication.

> We do already fill in the email address and username where they are available from the third party but we can't guarantee that the user won't need to change them - the username might already be in use for example.

We may have the screen to choose the username only if it's already used. Otherwise, they can change it in the settings later on.

> Also while we can trust the email if it's a gmail address from authenticating via google that's not really true for the other providers as they don't control the email addresses so at best they know that the address was valid at some point in the past, so we still need to do an email confirmation.

I believe it's safe for most of the providers already present, but if you think it's only ok for Google, I'm fine with that.

**Comment by tomhughes** [comment]
They all perform validation when you initially sign up with them - that's no guarantee the email is still working N years later when they signup with us.

It's not like we ignore the address - we use it to populate the signup form. We just don't suppress the confirmation check for all providers.

**Comment by tomhughes** [comment]
Note that we do also trust emails from facebook since #1096 though I'm pretty sure we're wrong to do that despite me giving in at the time. I'm sure all they mean by that documentation is that it was verified at some point in the past and to avoid the risk of being accused of spamming some innocent party that has taken over an old address we really need to verify that the current owner of the address wants to signup.

**Comment by ulope** [comment]
I can't recall any other service that offers social logins I've used in the past (and that are easily hundreds at this point) and still requires email verification.

The scenario you describe (users verifies email on 3rd party service, time passes, somehow loses / gives up that address, but still has access to the 3rd party account, email is reassigned to a new user) is so excessively unlikely and rare that it's not even worth considering. Much less inconveniencing everyone signing up on OSM.

In addition to that the fact that the confirmation email takes upwards of an hour to arrive doesn't help at all.

**Comment by tomhughes** [comment]
Well I'm sorry if not wanting to be seen as a source of spam inconveniences you but the consequences of various anti-spam organisations deciding we weren't validating emails before using them could be very serious in terms of getting our mail server blacklisted.

There's no reason it should take that long unless your provider is doing greylisting or something.

**Comment by ulope** [comment]
Well I'm giving you my perspective as a potential new user that has had a lot of experience using (and building) online web services.

What you're doing is driving away users. That's all I'm saying.

**Comment by woodpeck** [comment]
I think that we can afford to err on the side of diligence, rather than on the side of "making signup super quick". Ideally we're looking for people who will contribute a significant amount of their spare time over the next 10 years. The fact that we even allow third-party authentication is already more than the average web site has, we don't need to water that down further just to gain a few more clicks by people who cannot be bothered to fill something out.

**Comment by simonpoole** [comment]
Besides the fact that the sign up process is super simple, and if anything it is easier to sign up directly instead of using a third party service (probably asking for the e-mail address twice wouldn't really be necessary for 3rd party logins), the actual numbers indicate that the sign up rate has steadily increased over the last 19 years with no sign of abating.

I suspect the actual bone of contention is that we actually require a working e-mail address.

**Comment by ulope** [comment]
> I suspect the actual bone of contention is that we actually require a working e-mail address.

@simonpoole You suspect wrongly.

**Comment by francois2metz** [comment]
Closing as there is no change on any side.
